eox_core/api/v1/tests/test_grades.py

```diff
@@ -0,0 +1,150 @@
+    @patch_permissions
+    def test_get_grade_detail_no_policy(  # pylint: disable=too-many-arguments
+            self,
+            _,
+            get_edxapp_user,
+            get_enrollment,
+            grade_factory,
+            get_valid_course_key,
+            get_courseware_courses,
+    ):
+        """Test that the GET method works including section details"""
+
+        get_edxapp_user.return_value.username = "test"
+        get_enrollment.return_value = None, None
+        grade_factory.return_value.return_value.read.return_value.percent = 0.5
+        grade_factory.return_value.return_value.read.return_value.subsection_grades = {}
+        params = {
+            "course_id": "course-v1:org+course+run",
+            "username": "test",
+            "detailed": True,
+        }
+        expected_response = {"earned_grade": 0.5, "section_breakdown": []}
+
+        response = self.client.get(self.url, params)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data, expected_response)
+
+    @patch("eox_core.api.v1.views.get_courseware_courses")
+    @patch("eox_core.api.v1.views.get_valid_course_key")
+    @patch("eox_core.api.v1.views.get_course_grade_factory")
+    @patch("eox_core.api.v1.views.get_enrollment")
+    @patch("eox_core.api.v1.views.get_edxapp_user")
+    @patch_permissions
+    def test_get_grade_detail_policy(  # pylint: disable=too-many-arguments
+            self,
+            _,
+            get_edxapp_user,
+            get_enrollment,
+            grade_factory,
+            get_valid_course_key,
+            get_courseware_courses,
+    ):
+        """Test that the GET method works with section details and grading policy"""
+
+        get_edxapp_user.return_value.username = "test"
+        get_enrollment.return_value = None, None
+        grade_factory.return_value.return_value.read.return_value.percent = 0.5
+        grade_factory.return_value.return_value.read.return_value.subsection_grades = {}
+        get_courseware_courses.return_value.get_course_by_id.return_value.grading_policy = {
+            "GRADE_CUTOFFS": {}
+        }
+        params = {
+            "username": "test",
+            "course_id": "course-v1:org+course+run",
+            "grading_policy": True,
+            "detailed": True,
+        }
+        expected_response = {
+            "earned_grade": 0.5,
+            "grading_policy": {"grade_cutoffs": {}},
+            "section_breakdown": [],
+        }
+
+        response = self.client.get(self.url, params)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data, expected_response)
+
+    @patch("eox_core.api.v1.views.get_enrollment")
+    @patch("eox_core.api.v1.views.get_edxapp_user")
+    @patch_permissions
+    def test_get_grade_input_validation(self, _, get_edxapp_user, get_enrollment):
+        """ Test that the GET method requires some parameters """
```

Comment:
can you be a bit more specific with the description?

Reply:
Sure thing.